public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts

```diff
@@ -1530,6 +1542,104 @@ describe('ElasticResponse', () => {
   });
 });
 
+describe('addIdFieldToDataFrame', () => {
+  it('adds id field to the end of the field-list, if it does not exist yet', () => {
+    const df: DataFrame = {
+      fields: [
+        {
+          name: 'timestamp',
+          type: FieldType.time,
+          config: {},
+          values: [1, 2],
+        },
+        {
+          name: 'body',
+          type: FieldType.string,
+          config: {},
+          values: [1, 2],
+        },
+        {
+          name: 'hostname',
+          type: FieldType.string,
+          config: {},
+          values: ['host1', 'host2'],
+        },
+        {
+          name: '_id',
+          type: FieldType.string,
+          config: {},
+          values: ['id-1', 'id-2'],
+        },
+        {
+          name: '_index',
+          type: FieldType.string,
+          config: {},
+          values: ['i_d-x', 'i_dx-'],
+        },
+      ],
+      length: 2,
+    };
+
+    const mdf = new MutableDataFrame(df);
+
+    addIdFieldToDataFrame(mdf);
+
+    const lastField = mdf.fields.at(-1);
+    expect(lastField?.name).toBe('id');
+    expect(lastField?.type === FieldType.string);
+    expect(lastField?.values).toStrictEqual(['i__d_-x-id-1', 'i__dx_--id-2']);
+  });
+
+  it('does not add id field if it already exists', () => {
+    const df: DataFrame = {
+      fields: [
+        {
+          name: 'timestamp',
+          type: FieldType.time,
+          config: {},
+          values: [1, 2],
+        },
+        {
+          name: 'body',
+          type: FieldType.string,
+          config: {},
+          values: [1, 2],
+        },
+        {
+          name: 'id',
+          type: FieldType.string,
+          config: {},
+          values: ['x1', 'x2'],
+        },
+        {
+          name: '_id',
+          type: FieldType.string,
+          config: {},
+          values: ['id1', 'id2'],
+        },
+        {
+          name: '_index',
+          type: FieldType.string,
+          config: {},
+          values: ['idx', 'idx'],
+        },
+      ],
+      length: 2,
+    };
+
+    const mdf = new MutableDataFrame(df);
+    addIdFieldToDataFrame(mdf);
+
+    const idFields = mdf.fields.filter((f) => f.name === 'id');
+    expect(idFields).toHaveLength(1);
+    expect(idFields[0]?.values).toStrictEqual(['x1', 'x2']);
+  });
+
+  it('does not add id if not logs-request', () => {
+    // FIXME
```

Comment:
Is this done by 
```
it('should not have an id-field created', () => {
      const idFields = result.data[0].fields.filter((f) => f.name === 'id');
      expect(idFields).toHaveLength(0);
    });
```

or do you still plan to add it?

Reply:
good catch! will check that ...

Reply:
yes, that case is handled by the other test, removed this test.